esp-wifi/src/timer/mod.rs

```diff
@@ -22,24 +22,22 @@ use crate::TimeBase;
 
 pub(crate) static TIMER: Mutex<RefCell<Option<TimeBase>>> = Mutex::new(RefCell::new(None));
 
-pub fn setup_timer_isr(timebase: TimeBase) -> Result<(), esp_hal::timer::Error> {
```

Comment:
Imo these errors would be bugs in our deinit impl, therefore I've switched to just panicking instead. Users can't actually do anything if this fails.